core/node/node_sync/src/batch_transaction_updater/mod.rs

```diff
@@ -0,0 +1,302 @@
+                            continue;
+                        }
+                        Some(ref receipt) => {
+                            let sent_at_block: u32 = receipt.block_number.unwrap().as_u32();
+                            db_eth_tx_history.sent_at_block = Some(sent_at_block);
+                            connection
+                                .eth_sender_dal()
+                                .set_sent_at_block(db_eth_tx_history.id, sent_at_block)
+                                .await?;
+                            sent_at_block
+                        }
+                    }
+                }
+            };
+
+            // transaction as included, check if we can potentially update
+            // its finality status. This value is untrusted as seen_at_block may have been cached
+            let Some(finality_update) = l1_block_numbers
+                .get_finality_update(db_eth_tx_history.eth_tx_finality_status, sent_at_block)
+            else {
+                continue;
+            };
+
+            // we can potentially update finality status, but we need to
+            // validate all properties including block number as it may
+            // have been cached
+            let receipt = match receipt {
+                Some(receipt) => receipt,
+                None => {
+                    let Some(receipt) =
+                        self.sl_client.tx_receipt(db_eth_tx_history.tx_hash).await?
+                    else {
+                        tracing::warn!(
+                            "Expected transaction {} to be mined at block {}, but transaction not mined at all on SL",
+                            db_eth_tx_history.tx_hash,
+                            db_eth_tx_history.sent_at_block.unwrap()
+                        );
+                        connection
+                            .eth_sender_dal()
+                            .unset_sent_at_block(db_eth_tx_history.id)
+                            .await?;
+                        continue;
+                    };
+                    receipt
+                }
+            };
+
+            let db_eth_history_id = db_eth_tx_history.id;
+            let result = self
+                .apply_status_update(
+                    &mut connection,
+                    db_eth_history_id,
+                    receipt,
+                    &l1_block_numbers,
+                )
+                .await;
+
+            if result.is_ok() {
+                updated_count += 1;
+            } else {
+                tracing::warn!(
+                    "Failed to update finality status for transaction {} with type {} from {:?} to {:?} due to error: {}",
+                    db_eth_tx_history.tx_hash,
+                    db_eth_tx_history.tx_type,
+                    db_eth_tx_history.eth_tx_finality_status,
+                    finality_update,
+                    result.err().unwrap()
```

Comment:
Nit: you can use `if let Err(_) = result { .. } else { updated_count += 1 }` to avoid `unwrap()`s. But more generally, swallowing *all* errors looks incorrect (see the other comment).